### PR TITLE
Don't return direct link to FxA for login/register, use our API endpoint

### PR DESCRIPTION
**Pull request by diox**
This allows us to avoid modifying the session on pages that just want to display a login/register link, which in turn avoids creating a new session and potentially returning a session cookie to anonymous users.

Fixes https://github.com/mozilla/addons-server/issues/18556

### CODE REVIEW

File: src/olympia/accounts/templatetags/jinja_helpers.py [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 @jinja2.pass_context
 def register_link(context):
-    return utils.default_fxa_register_url(context['request'])
-
-
-@library.global_function
-@jinja2.pass_context
-def fxa_config(context):
-    return utils.fxa_config(context['request'])
+    return login_link(context)
```
Comment (bobsilverberg) [inline src/olympia/accounts/templatetags/jinja_helpers.py:21]: If `register_link` just returns `login_link` now, do we need `register_link` at all? Could we just call `login_link` from anywhere we're currently calling `register_link`? If so, should it perhaps be called `fxa_link`?

File: src/olympia/accounts/tests/test_helpers.py [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
 
-@mock.patch(
-    'olympia.accounts.templatetags.jinja_helpers.utils.default_fxa_login_url',
-    lambda c: 'http://auth.ca',
-)
 def test_login_link():
     request = RequestFactory().get('/en-US/firefox/addons')
-    assert jinja_helpers.login_link({'request': request}) == ('http://auth.ca')
+    assert jinja_helpers.login_link({'request': request}) == (
+        'http://testserver/api/v5/accounts/login/start/'
```
Comment (bobsilverberg) [inline src/olympia/accounts/tests/test_helpers.py:9]: Would it be worth putting `'http://testserver/api/v5/accounts/login/start/'` into a constant as it's used in each of the three assertions below? Also, how do we know that is the correct URL to expect? Can that be imported from elsewhere to be sure it's the correct value?

File: src/olympia/accounts/utils.py [hunk 3]
```diff
@@ -18,28 +18,6 @@
 from .tasks import clear_sessions_event, delete_user_event, primary_email_change_event
 
 
-def fxa_config(request):
```
Comment (bobsilverberg) [inline src/olympia/accounts/utils.py:21]: Just curious, with the removal of this, what was it ever used for?

File: src/olympia/amo/decorators.py [hunk 4]
```diff
@@ -181,22 +181,3 @@ def wrapper(request, *args, **kw):
         return response
 
     return wrapper
-
-
-def allow_mine(f):
```
Comment (bobsilverberg) [inline src/olympia/amo/decorators.py:186]: What does this have to do with this patch?

**Review by bobsilverberg — APPROVED** [review]
Looks good. I just have a few questions and a suggestion.

r+wc

File: src/olympia/amo/decorators.py (hunk 4 above)
Comment (diox) [reply src/olympia/amo/decorators.py:186]: Just a drive-by cleanup while I was looking at this file for stuff to change around auth

File: src/olympia/accounts/utils.py (hunk 3 above)
Comment (diox) [reply src/olympia/accounts/utils.py:21]: It was used in the `fxa_config` helper which was called from the base template to add a `data-` attribute containing that info. That would presumably have been useful a long time ago with the legacy site.

File: src/olympia/accounts/templatetags/jinja_helpers.py (hunk 1 above)
Comment (diox) [reply src/olympia/accounts/templatetags/jinja_helpers.py:21]: We technically don't need it anymore, I wasn't sure if we'd need the distinction. I guess I can remove it and rename the other.

File: src/olympia/accounts/tests/test_helpers.py (hunk 2 above)
Comment (diox) [reply src/olympia/accounts/tests/test_helpers.py:9]: If you don't hardcode the expected result, then you end up re-creating the logic from the function being tested (which is super short - it's just concatenating the result of `drf_url()` with `path_with_query()` encoded).

I could add a constant, but I'm not super convinced it's necessary since it's the same block of code and it's in a test.

Comment (bobsilverberg) [reply src/olympia/accounts/tests/test_helpers.py:9]: > If you don't hardcode the expected result, then you end up re-creating the logic from the function being tested (which is super short - it's just concatenating the result of `drf_url()` with `path_with_query()` encoded).
> 
> I could add a constant, but I'm not super convinced it's necessary since it's the same block of code and it's in a test.

Fair enough. We tend to use constants whenever possible in tests for frontend, but I know that's not the case for server, so this is consistent with the way a lot of other tests are written.

File: src/olympia/accounts/templatetags/jinja_helpers.py (hunk 1 above)
Comment (diox) [reply src/olympia/accounts/templatetags/jinja_helpers.py]: 450ea6c7e8870a81750ccd50e94849541d96634c (didn't end up doing the renaming, I don't think it's that much clearer)
